lib/ronin/recon/values/url.rb

```diff
@@ -211,6 +215,36 @@ def self.value_type
           :url
         end
 
+        #
+        # Case equality method usef for fuzzy matching
+        #
+        # @param [Wildcard, Domain, Host, URL, Value] other
+        #   The other value to compare.
+        #
+        # @return [Boolean]
+        #   Imdicates whether the other value is either a {Domain} and has the
+        #   same domain name, or a {Host} and shares the same domain name.
+        #
+        def ===(other)
+          case other
+          when URL
+            uri == other.uri
+          when Domain
+            host == other.name || host.end_with?(".#{other.name}")
+          when Website
+            scheme == other.scheme &&
+              host == other.host &&
+              port == other.port
+          when Wildcard
+            prefix, suffix = other.template.split('*',2)
+
+            host.start_with?(prefix) && host.end_with?(suffix)
+          when Host
+            host == other.name
```

Comment:
Should `Domain`, `Website`, `Wildcard`, `Host` be comparable to a `URL`? Or should only other `URL`s be compared to the `URL`? Would it make sense for a user to specify `ronin-recon run DOMAIN WILDCARD URL1 URL2` and the recon engine would end up scanning/spidering the host names of the URLs?